content-src/components/Spotlight/Spotlight.js

```diff
@@ -22,26 +27,31 @@ const SpotlightItem = React.createClass({
 
 SpotlightItem.propTypes = {
   url: React.PropTypes.string.isRequired,
-  image: React.PropTypes.string.isRequired,
-  icon: React.PropTypes.string.isRequired,
-  title: React.PropTypes.string.isRequired,
+  images: React.PropTypes.array,
+  favicon_url: React.PropTypes.string,
+  icons: React.PropTypes.array,
+  title: React.PropTypes.string,
   description: React.PropTypes.string.isRequired,
 };
 
 const Spotlight = React.createClass({
+  getDefaultProps() {
+    return {length: DEFAULT_LENGTH};
+  },
   render() {
-    const {props} = this;
+    const sites = this.props.sites.slice(0, this.props.length);
```

Comment:
This truncates `this.props.sites` to `this.props.length`, not `this.props.sites.length`